std/range/package.d

```diff
@@ -10899,7 +10899,15 @@ cost $(BIGOH n), use $(REF isSorted, std,algorithm,sorting).
 auto assumeSorted(alias pred = "a < b", R)(R r)
 if (isInputRange!(Unqual!R))
 {
-    return SortedRange!(Unqual!R, pred)(r);
+    // Avoid senseless `SortedRange!(SortedRange!(...), pred)` nesting.
+    static if (is(R == SortedRange!Args, Args...))
+    {
+        return SortedRange!(Unqual!(typeof(r._input)), pred)(r._input);
```

Comment:
why not add another branch which just returns the input if the `pred` is the same when checked by `__traits(isSame)`?

Reply:
Added.